job-dsl-plugin/src/main/groovy/javaposse/jobdsl/plugin/ExecuteDslScripts.java

```diff
@@ -473,10 +481,25 @@ private void updateGeneratedJobs(final Job seedJob, TaskListener listener,
         // print what happened with unreferenced jobs
         logItems(listener, "Disabled items", disabled);
         logItems(listener, "Removed items", removed);
+        logItems(listener, "Shelved items", shelved);
 
         updateGeneratedJobMap(seedJob, Sets.union(added, existing), unreferenced);
     }
 
+    private void shelve(Item project) {
+       Jenkins jenkins = Jenkins.get();
+       if (! (project instanceof BuildableItem)) {
+           LOGGER.log(Level.WARNING, "Unable to shelve " + project + " since it is not a BuildableItem");
+           return;
+       }
+       BuildableItem item = (BuildableItem) project;
+        if (jenkins.getPlugin(SHELVE_PLUGIN_ID) == null) {
+            LOGGER.log(Level.WARNING, "Unable to shelve project " + item + " since the " + SHELVE_PLUGIN_ID + " plugin is not installed.");
+            return;
+        }
+        jenkins.getQueue().schedule(new ShelveProjectTask(item), 0);
```

Comment:
`ShelveProjectTask` does not seem to check if the user has permission to delete a project. That is done upfront in `ShelveProjectAction`. So the permission must be checked here.

Reply:
Done.